Assignment2/Q1.java

```diff
@@ -0,0 +1,148 @@
+//The tree created is a binary tree to match the example given but the method does not make any use of binary tree features 
```

Comment:
1 General comment. Both of your solutions are recursive (which is fine). But it would be a good learning exercise to try and write a non-recursive solution. Unlike functional programming languages, recursive solutions can be hard to argue about space efficiency in Java. By making the solution non-recursive, you avoid stackoverflows, and can make a good space argument.

This isn't required, but knowing how to do it and understnading the pros and cons of recursive and non-recursive solutions is a big positive in Google interviews.